src/components/App.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react'
+// IMPORT SEMANTIC UI STYLE BASIS
+import 'semantic-ui-css/semantic.min.css'
+// IMPORT SEMANTIC-UI-REACT COMPONENTS
+import { Container } from 'semantic-ui-react'
+// IMPORT API
+import api from '../apis/user'
+// IMPORT COMPONENTS
+import UserInfo from './UserInfo'
+import Login from './Login'
+import CreateUser from './CreateUser'
+// IMPORT STORAGE HELPER
+import storage from '../helpers/storage'
+
+class App extends React.Component {
+  state = {
+    user: null,
+    createUser: false,
+    isLoading: false
+  }
+
+  componentDidMount() {
+    const userLogged = storage.getItem('logged')
+    userLogged && this.setState({ user: userLogged })
+  }
+
+  logUser = async user => {
+    if (!this.state.isLoading) {
+      this.setState({ isLoading: true })
+
+      const result = await api.login(user)
+
+      this.setState({ isLoading: false })
+
+      if (result.status === 200) {
+        const { name, email } = result.data.userLogged
+        const { token } = result.data
+
+        this.setState({ user: { name, email, token } })
```

Comment:
Also put it in the else.

```suggestion
        this.setState({ user: { name, email, token }, isLoading: false })
```

Another option is to save user in variable. Be sure to not remove values that are already in user. Remember that the idea is to avoid re-rendering and recheck between the dom and shadow dom.